lib/web/ui/server.go

```diff
@@ -143,15 +145,28 @@ func MakeKubeClusters(clusters []types.KubeCluster) []KubeCluster {
 
 		sort.Sort(sortedLabels(uiLabels))
 
+		kubeUsers, kubeGroups := userRoles.EnumerateKubernetesUsersAndGroups(cluster)
+
 		uiKubeClusters = append(uiKubeClusters, KubeCluster{
-			Name:   cluster.GetName(),
-			Labels: uiLabels,
+			Name:       cluster.GetName(),
+			Labels:     uiLabels,
+			KubeUsers:  nonNilSlice(kubeUsers.Allowed()),
+			KubeGroups: nonNilSlice(kubeGroups.Allowed()),
 		})
 	}
 
 	return uiKubeClusters
 }
 
+// nonNilSlice checks if the given slice is nil. If it's the case returns an empty
+// slice, otherwise returns the given slice.
+func nonNilSlice(s1 []string) []string {
```

Comment:
Why is this needed?

Reply:
I followed the logic in SSH logins where non-nil arrays are returned. The same applies to `omitempty` where our API does not have any  `omitempty` for teleport resources such as `Database`, `Server`, `Kubes`, and `DesktopService`.

@kimlisa do you have any intel for this?

Reply:
i don't know the origins unfortunately, and in case of webapps, it's not required. Maybe the person who created it knew that nil slices are encoded as `null` versus an empty array. But a non-nil slice is encoded as empty array? But the frontend does not expect the backend to explicitly set it to an empty array for us, since we can't depend/assume it's never null/undefined.